### DISCUSSION TITLE
[A-L] (12 Gennaio 2022) Esercizio 4

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#323] [Esami A-L]
![image](https://user-images.githubusercontent.com/69389461/152745978-f936ae39-6089-4cc4-aa18-ac6282e3c60c.png)

**Comment by i0nut02** [comment]
<img width="399" alt="Schermata 2022-02-07 alle 09 54 19" src="https://user-images.githubusercontent.com/99051485/152755971-ed548fe8-6ba8-4d06-ad06-d6b9ea9425d3.png">

**Comment by AxnNxs** [comment]
> <img alt="Schermata 2022-02-07 alle 09 54 19" width="399" src="https://user-images.githubusercontent.com/99051485/152755971-ed548fe8-6ba8-4d06-ad06-d6b9ea9425d3.png">

Caso Base: 
non è necessario che a e b siano uguali a 0, ma è solo un caso particolare.
Quanto n = 0, che consiste già nel caso base, l'equazione diventa parametrica in a e b, trasformandosi in:

0 = 3a + b            da cui
b = - 3a                il che rappresenta un insieme di soluzioni e non una sola. 

In particolare abbiamo che:

b = 0   ^   b = 1   ^   b = 2            da cui
0 = -3 x (0)   ^   1 = -3 x ( -1/3 )   ^   2 = -3 x ( -2/3 )          con  "a" appartenente all'insieme dei Reali.


Passo induttivo:
Se P(n)   =>    P(n+1), dove P(n) è l'Ipotesi induttiva per qualche n >= 0.
Pertanto, sapendo che n = 3a + b, abbiamo che:

n+1 = 3a + b      =>      0 + 1 = 3a + b       da cui:
1 = 3a + b      da cui:
b = 1 - 3a      da cui:
0 = 1 - 3 x (1/3)   ^   1 = 1 - 3 x (0)   ^   2 = 1 - 3 x ( - 1)         al variare di "a" nei Reali.




CVD (in attesa di ulteriori verifiche)

**Comment by Elia-Belli** [comment]
@AxnNxs $a \in \mathbb N$ quindi non può variare nei Reali.

- **Caso Base**: $n=0$ $0 = 3a+b$ è soddisfatta da $a = 0 \land b=0$
- **Passo Induttivo**: $n \geq 0$  $P(n) \Rightarrow P(n+1)$
$P(n): n = 3a+b$
$P(n+1): n+1 = 3a + b \rightarrow n = 3a + b -1$

Io l'ho interpretata in questo modo: ponendo $b=1$ possiamo ottenere tutti i multipli di 3, a questo punto dobbiamo "coprire" anche i 2 numeri compresi tra ogni multiplo di 3, possiamo ottenere il successivo ponendo $b = 2$ ottenendo l'operazione di $+1$ e l'antecedente ponendo $b=0$ ottenendo l'operazione di $-1$.

**Comment by i0nut02** [comment]
di base nel passo induttivo mi sono sbagliato a dire che n +1 = 3a + b ma in realtà sarebbe n+1 = 3a' + b', allora parto dicendo 
n +1 = 3a + b + 1 e da qui divido tutto in casi e assegno ad a' e b' i valori in base ai vari casi

**Comment by aflaag** [comment]
Il **caso base** si ha per $n = 0$, per il quale banalmente esistono $a = b = 0$ per cui $0 = 3 \cdot 0 + 0$.

Per **ipotesi induttiva**, assumiamo che per $n > 0$ esistano $a, b \in \mathbb N$ tali che $n = 3a + b$ e $0 \le b \le 2$.

Allora, nel **passo induttivo** dobbiamo dimostrare che per $n + 1$ esistono ancora due valori $a', b' \in \mathbb N$ tali che $n + 1 = 3a' + b'$ e $0 \le b' \le 2$. Per ipotesi induttiva sappiamo che $n = 3a + b$, dunque possiamo riscrivere l'equazione come $3a + b + 1 = 3a' + b'$, e poiché $0 \le b \le 2$, possono verificarsi 3 casi:
- $b = 0$, allora $n = 3a$ e dunque $n + 1 = 3a' + b' \implies 3a + 0 + 1 = 3a' + b' \iff a' = a \land b' = 1$
- $b = 1$, allora $n = 3a + 1$ e dunque $n + 1 = 3a' + b' \implies 3a + 1 + 1 = 3a' + b' \iff a' = a \land b' = 2$
- $b = 2$, allora $n = 3a + 2$ e dunque $n + 1 = 3a' + b' \implies 3a + 2 + 1 = 3a' + b'$ ma non possiamo porre $b' = 3$ perché $b'$ deve ancora essere tale che $0 \le b' \le 2$; notiamo però che $3a + 2 + 1 = 3a + 3 = 3(a + 1)$ e dunque $3(a + 1) = 3a' + b' \iff a' = a + 1 \land b'= 0$

**Comment by DavideGalilei** [comment]
Dimostrazione poco formale (ma per capire il meccanismo):

![image](https://github.com/user-attachments/assets/f81e5d2d-d48e-474a-a022-cd7aec817fe9)
